### PR TITLE
feat: adding metrics for signature decoding

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Adding metrics for signature decoding.

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3644

## **Manual testing steps**

1. Enable signature deccoding locally
2. Open permit page
3. Check that metrics are recorded for signature decoding information

## **Screenshots/Recordings**
NA

## **Pre-merge author checklist**

- [X] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [X] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts:1]: As this isn't related to transaction simulation, should we move this hook to a signature specific folder or just `ui/pages/confirmations/hooks`?

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { useEffect } from 'react';
+import { DecodingDataStateChange } from '@metamask/signature-controller';
+
+import { SignatureRequestType } from '../../types/confirm';
+import { useConfirmContext } from '../../context/confirm';
+import { useSignatureEventFragment } from '../../hooks/useSignatureEventFragment';
+
+enum DecodingResponse {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts:8]: Minor, `DecodingResponseType`?

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    renderHookWithConfirmContextProvider(
+      () => useDecodedSignatureMetrics(),
+      state,
+    );
+
+    expect(mockUpdateSignatureEventFragment).toHaveBeenCalledTimes(1);
+    expect(mockUpdateSignatureEventFragment).toHaveBeenLastCalledWith({
+      properties: {
+        decoding_change_types: [],
+        decoding_response: 'NO_CHANGE',
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts:52]: Do we also need unit tests for the change and error types?

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    renderHookWithConfirmContextProvider(
+      () => useDecodedSignatureMetrics(),
+      state,
+    );
+
+    expect(mockUpdateSignatureEventFragment).toHaveBeenCalledTimes(1);
+    expect(mockUpdateSignatureEventFragment).toHaveBeenLastCalledWith({
+      properties: {
+        decoding_change_types: [],
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts:51]: And a test that includes change types?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/pages/confirmations/hooks/useDecodedSignatureMetrics.test.ts [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+    expect(mockUpdateSignatureEventFragment).toHaveBeenCalledTimes(1);
+    expect(mockUpdateSignatureEventFragment).toHaveBeenLastCalledWith({
+      properties: {
+        decoding_change_types: [],
+        decoding_response: 'NO_CHANGE',
+      },
+    });
+  });
+
+  it('should call updateSignatureEventFragment with correct parameters if there are no state changes', async () => {
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/hooks/useDecodedSignatureMetrics.test.ts:71]: Minor, same name as the other?

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts (hunk 3 above)
Comment (jpuri) [reply ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts]: done

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts (hunk 1 above)
Comment (jpuri) [reply ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.ts]: done

File: ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts (hunk 2 above)
Comment (jpuri) [reply ui/pages/confirmations/components/simulation-details/useDecodedSignatureMetrics.test.ts]: done

File: ui/pages/confirmations/hooks/useDecodedSignatureMetrics.test.ts (hunk 4 above)
Comment (jpuri) [reply ui/pages/confirmations/hooks/useDecodedSignatureMetrics.test.ts]: fixed

**Review by matthewwalsh0 — APPROVED** [review]

**Review by digiwand — APPROVED** [review]

**Pull request merged**